roles/tests-integration/tests/common/sniffer.rs

```diff
@@ -50,17 +52,56 @@ pub struct Sniffer {
     upstream_address: SocketAddr,
     messages_from_downstream: MessagesAggregator,
     messages_from_upstream: MessagesAggregator,
+    intercept_messages: Vec<InterceptMessage>,
```

Comment:
we have some comments above `Sniffer` declaration explaining its internal details

we should make sure those comments also include this new funcitonality